### PR TITLE
Wait for SR-IOV networks to be allocatable on the worker nodes

### PR DESCRIPTION
**Pull request by tkrishtop**
build-depends: https://github.com/dci-labs/inventories/pull/76

### CODE REVIEW

File: testpmd/hooks/files/collect-sriov-operator-data.sh.j2 [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
```
Comment (tonyskapunk) [inline testpmd/hooks/files/collect-sriov-operator-data.sh.j2:1]: I guess this needs to be renamed to be used as a script and not as a template, right?

File: testpmd/hooks/pre-run.yml [hunk 2]
```diff
@@ -85,6 +85,11 @@
     loop_var: sriov
   when: sriov_network is defined
 
+- name: SR-IOV must-gather
+  shell: >
+    bash files/collect-sriov-operator-data.sh | tee {{ job_logs }}/collect-sriov-operator-data.txt
```
Comment (tonyskapunk) [inline testpmd/hooks/pre-run.yml:90]:
```suggestion
    bash files/collect-sriov-operator-data.sh | tee {{ job_logs.path }}/collect-sriov-operator-data.txt
```

Comment (tkrishtop) [reply testpmd/hooks/pre-run.yml]: thank you @tonyskapunk , fixed both :)

**Review by manurodriguez — APPROVED** [review]
Thanks Tatiana and Tony, this will help us get sr-iov details without having to check must-gather :)

File: testpmd/hooks/pre-run.yml [hunk 3]
```diff
@@ -85,6 +85,28 @@
...
+    oc get nodes --no-headers=true --selector='!node-role.kubernetes.io/master' | wc -l
+  register: count_workers
+
+# TODO: add rescue in the case of fail
+- name: Wait to have all networks allocatable on worker nodes
+  shell: >
+    oc get nodes --selector='!node-role.kubernetes.io/master' -o jsonpath='{.items[*].status.allocatable}'
+  register: nodes_allocation
+  until:
+    - nodes_allocation.stdout | regex_findall(sriov_networks[0].resource) | length == count_workers.stdout | int
```
[the remark this answers is not in the document]
Comment (tkrishtop) [reply testpmd/hooks/pre-run.yml]: Hi @manurodriguez thank you! I fixed the path and var name, and rerun the job. Let's see :)

**Comment by tkrishtop** [comment]
To wait for having networks allocatable on the worker nodes [worked after 2 min](https://www.distributed-ci.io/jobs/33f52290-5b5c-4edd-ae7c-252a7b067134/jobStates#2f368d0d-a634-4600-8d34-9de6fbd7e043:file73). We're a bit on the limit though, so I'll increase waiting time a bit.

[SR-IOV must-gather worked fine as well](https://www.distributed-ci.io/files/64f924f2-c95c-426d-9d18-08dacaa9b1c7).

Unfortunately, the job failed with a side error before starting the loadbalancer pod, so I'll re-run it.

Todo:
1/ Increase wait time for networks to be allocatable on the workers up to 5 min
2/ Add rescue if networks are failed to be allocated: delete all sriov-network-config-daemon-XXXXX pods and then wait for 3 min to have them restarted.

**Comment by tkrishtop** [comment]
1/ To wait for SR-IOV networks creation works - [we see that first networks are not available in allocatable resources of the workers, and after 2 min everything is fine](https://www.distributed-ci.io/jobs/3686f332-d3eb-4a31-9dce-d7db5c7822b3/jobStates#175b6ec4-dc64-4bc8-96dc-f3b49e0d9151:file73).

2/ SR-IOV must-gather works fine as well, [here is an output](https://www.distributed-ci.io/files/0b204a71-cd55-4026-b387-38f3774a584d).

The change is already safe to merge.

TODO: Add rescue if networks are failed to be allocated: delete all sriov-network-config-daemon-XXXXX pods and then wait for 3 min to have them restarted.

P.S. All jobs are now failing because of another issue - packagemanifests creation, and we couldn't check the status of the loadbalancer pod.

**Comment by tkrishtop** [comment]
The wait workaround in the last job didn't work: [only three workers out of four had SR-IOV networks when the wait time expired](https://www.distributed-ci.io/jobs/2e4c5b99-9e90-4f7f-8c57-5a2cce03c96b/jobStates#dd624aa4-4f11-4b3d-891a-6611ea08a0a4:file73). Worker-1 [didn't have SR-IOV among allocatable](https://www.distributed-ci.io/files/7f54bbe6-670b-4bc1-bd98-4990ee60ccaf) because it was somehow unschedulable:
```
    name: worker-1
    taints:
    - effect: NoSchedule
      key: node.kubernetes.io/unschedulable
    unschedulable: true
  status:
    allocatable:
      cpu: "72"
      devices.kubevirt.io/kvm: "110"
      devices.kubevirt.io/tun: "110"
      devices.kubevirt.io/vhost-net: "110"
      ephemeral-storage: 1830801Mi
      hugepages-1Gi: 64Gi
      hugepages-2Mi: 48000Mi
      memory: 410621136Ki
      pods: "250"
```

To compare with the [good job](https://www.distributed-ci.io/jobs/3686f332-d3eb-4a31-9dce-d7db5c7822b3/jobStates#175b6ec4-dc64-4bc8-96dc-f3b49e0d9151:file73).

Next steps:
- Double wait time
- Introduce rescue: delete all sriov-network-config-daemon-XXXXX pods, and then repeat wait for networks to be ready

**Comment by tkrishtop** [comment]
[Wait to allocate SR-IOV works fine in OCP-4.8](https://www.distributed-ci.io/jobs/49c13a3c-5959-48c0-9dcd-85c59e5903d6/jobStates#0bb79b23-32a9-4da0-bd73-cbec695fe584:file74).

But then [loadbalancer pod = Pending, container = ContainerCreating](https://www.distributed-ci.io/jobs/49c13a3c-5959-48c0-9dcd-85c59e5903d6/jobStates#d19764e1-20e1-41d6-9789-13754e9cec41:file36) because I forgot to deactivate deps with PR#242.

Finally, TRex tests are passing OK, and in the example-cnf_status.log we see loadbalancer pod running
```
loadbalancer-f5b6c598d-vc2tx                               2/2     Running
```

and [its logs look perfectly fine](https://www.distributed-ci.io/files/6c1c7a65-0e06-4da3-8eb8-5869c13f2c47).

I will deactivate PR#242 and re-run the tests once again.

**Comment by manurodriguez** [comment]
Tatiana, sure, let's see how it goes this last test, I re-schedule it because last one failed for some catalog stuff

**Pull request merged**